### PR TITLE
Update Developer.yaml and add Dragonfly DB

### PR DESCRIPTION
**Pull request by magedragheb**
Update Developer.yaml and add Dragonfly DB, it is a redis compatible In-Memory datastore. The company is located in Israel, occupied Palestine.

### CODE REVIEW

File: dataset/companies/Developer.yaml [hunk 1]
```diff
@@ -82,3 +82,7 @@
   Description: ""
   Website: "https://ai21.com/"
   Alternatives: []
+- Name: "Dragonfly DB"
+  Description: "A redis compatible In Memory datastore"
+  Website: "https://www.dragonflydb.io/"
+  Alternatives: ["Memcached", "KeyDB"]
```
Comment (abarmawi) [inline dataset/companies/Developer.yaml:88]: We have introduced a schema for the Alternatives, Please check the README file and update this

**Review by abarmawi — CHANGES_REQUESTED** [review]

File: dataset/companies/Developer.yaml (hunk 1 above)
Comment (magedragheb) [reply dataset/companies/Developer.yaml]: Done, please review.

File: dataset/companies/Developer.yaml [hunk 2]
```diff
@@ -82,3 +82,21 @@
   Description: ""
   Website: "https://ai21.com/"
   Alternatives: []
+- Name: "Dragonfly DB"
+  Description: "A redis compatible In Memory datastore"
+  Website: "https://www.dragonflydb.io/"
+  Alternatives:
```
Comment (abarmawi) [inline dataset/companies/Developer.yaml:88]: let's stick to yaml format. Thanks a lot

```
Alternatives:
 - Name: ...
   Website: ...
```

File: package-lock.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
       "version": "8.12.0",
```
Comment (abarmawi) [inline package-lock.json:18]: Can you exclude this change?

**Pull request closed without merging**